From be7c7e74d7982ebc390ba4436e9513bbea791a33 Mon Sep 17 00:00:00 2001
From: Hans-Kristian Arntzen <post@arntzen-software.no>
Date: Mon, 21 Sep 2026 19:17:49 +0200
Subject: [PATCH 11/19] tests: Only apply the small wait on vkd3d-proton.

The sleep "breaks" AMD, because it copies too fast
for the test to verify racy signal. Sleeps on windows
are very low resolution by default, and can't be arsed
to hack that through just for a test.

Signed-off-by: Hans-Kristian Arntzen <post@arntzen-software.no>
---
 tests/d3d12_sync.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/tests/d3d12_sync.c b/tests/d3d12_sync.c
index 2c99e954..34084ba0 100644
--- a/tests/d3d12_sync.c
+++ b/tests/d3d12_sync.c
@@ -2335,8 +2335,13 @@ static void test_fence_signal_order_raced_signal_inner(bool use_shared)
         ID3D12CommandQueue_Signal(queue_a, fence_alt, 1);
         ID3D12CommandQueue_Signal(queue_a, fence, 2);
 
-        /* Make it more likely that signal order for fence = 2 is lower than fence = 1. */
-        vkd3d_sleep(1);
+        if (is_vkd3d_proton_device(context.device))
+        {
+            /* Make it more likely that signal order for fence = 2 is lower than fence = 1.
+             * We have submission threads, and to make the test more useful we
+             * should make it more likely that submission thread submits in order too. */
+            vkd3d_sleep(1);
+        }
 
         /* Deliberately test racy signal.
          * We expect fence = 2 to be signaled after fence = 1 despite being submitted first.
